### DISCUSSION TITLE
How to `PointToScreen`?

### DISCUSSION THREAD

**Discussion by lindexi** [unoplatform/uno#17196] [Q&A]
We can do PointToScreen in WinForm or WPF by https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.control.pointtoscreen?view=windowsdesktop-8.0 and https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.visual.pointtoscreen?view=windowsdesktop-8.0

But I can not find the equivalent of method in UNO.

And the https://stackoverflow.com/a/38580862 does not seem to help.

**Comment by carldebilly** [comment]
To transform the coordinates of an element relative to another in Uno Platform (and WinUI/UWP...), you can use the `.TransformToVisual()` method. This method creates a transformation that can be applied to convert the coordinates of a source element into the coordinates of a target element.

You _Stack Overflow_ article is the right way to go.

``` csharp
// Assume sourceElement and targetElement are your UIElements
var transform = sourceElement.TransformToVisual(targetElement);

// Local coordinates in sourceElement
var pointInSourceElement = new Point(x, y);

// Coordinates in targetElement
var pointInTargetElement = transform.TransformPoint(pointInSourceElement);

// pointInTargetElement now contains the coordinates in the context of targetElement
```

**Reply by lindexi** [reply]
@carldebilly But... it not the screen coordinate

**Reply by jeromelaban** [reply]
There's no API to get the screen coordinates at this time, you'll have derive it from the window position and the `TransformToVisual`.

**Reply by lindexi** [reply]
How can I get the windows from UIElement?

**Reply by MartinZikmund** [reply]
The API to retrieve position of the window is in progress, will get the PR finalized soon

**Reply by lindexi** [reply]
Thank you @MartinZikmund
